code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -374,6 +374,17 @@
 		return
 	if(default_deconstruction_crowbar(I))
 		return
+	if(panel_open)
+		if(istype(I, /obj/item/stock_parts/cell))
+			I.forceMove(src) // Force it out of our hands so we can put the old cell in it
```

Comment:
This should use a corresponding inventory proc (`trransferItemToLoc`), so it accounts for nodrop or w/e. 
```dm
if(!user.transferItemToLoc(I, src))
    return
```